contracts/interfaces/ITemplate.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.24;
+
+import '@openzeppelin/contracts/token/ERC721/IERC721.sol';
+
+/**
+ * @title ITemplate Interface
+ * @dev Interface for the Template contract that manages predefined templates
+ * for SACD creation with bit arrays and IPFS URLs
+ */
+interface ITemplate is IERC721 {
+  struct TemplateData {
```

Comment:
If we define `structs`, `events` and `errors` here, we can just inherit this interface in the `Template.sol`